XLS-0066d-lending-protocol/README.md

```diff
@@ -0,0 +1,1560 @@
+| `OverpaymentInterestRate` |    `No`     | :heavy_check_mark: | `number`  |   `UINT16`    |                        `N/A`                        | An interest rate charged on overpayments in 1/10th basis points. Valid values are between 0 and 100000 inclusive. (0 - 100%)                                   |
+| `StartDate`               |    `No`     | :heavy_check_mark: | `number`  |   `UINT32`    |                        `N/A`                        | The timestamp of when the Loan starts [Ripple Epoch](https://xrpl.org/docs/references/protocol/data-types/basic-data-types/#specifying-time).                  |
+| `PaymentInterval`         |    `No`     | :heavy_check_mark: | `number`  |   `UINT32`    |                        `N/A`                        | Number of seconds between Loan payments.                                                                                                                       |
+| `GracePeriod`             |    `No`     | :heavy_check_mark: | `number`  |   `UINT32`    |                        `N/A`                        | The number of seconds after the Payment Due Date that the Loan can be Defaulted.                                                                               |
+| `PreviousPaymentDate`     |    `N/A`    | :heavy_check_mark: | `number`  |   `UINT32`    |                         `0`                         | The timestamp of when the previous payment was made in [Ripple Epoch](https://xrpl.org/docs/references/protocol/data-types/basic-data-types/#specifying-time). |
+| `NextPaymentDueDate`      |    `N/A`    | :heavy_check_mark: | `number`  |   `UINT32`    |    `LoanSet.StartDate + LoanSet.PaymentInterval`    | The timestamp of when the next payment is due in [Ripple Epoch](https://xrpl.org/docs/references/protocol/data-types/basic-data-types/#specifying-time).       |
+| `PaymentRemaining`        |    `N/A`    | :heavy_check_mark: | `number`  |   `UINT32`    |               `LoanSet.PaymentTotal`                | The number of payments remaining on the Loan.                                                                                                                  |
+| `AssetAvailable`          |    `N/A`    | :heavy_check_mark: | `number`  |   `NUMBER`    | `LoanSet.[PrincipalRequested - LoanOriginationFee]` | The asset amount that is available in the Loan.                                                                                                                |
+| `PrincipalOutstanding`    |    `No`     | :heavy_check_mark: | `number`  |   `NUMBER`    |            `LoanSet.PrincipalRequested`             | The principal amount requested by the Borrower.                                                                                                                |
```

Comment:
```suggestion
| `PrincipalOutstanding`    |    `Yes`     | :heavy_check_mark: | `number`  |   `NUMBER`    |            `LoanSet.PrincipalRequested`             | The principal amount requested by the Borrower.                                                                                                                |
```
This is used to track the balance still owed, and is referenced in several places as being changed. In that case, it should be modifiable.

Reply:
Modifiable column refers to whether a user can directly modify the field. For example, MPTokenMetadata in the MPTokenIssuance object.

Reply:
I see. In that case, I think it would make more sense to rename the column(s) to "User modifiable?" or "Directly modifiable?" or something like that.

It's worth noting, though, I think that all the other fields that are not user modifiable are also not "automatically" modifiable.

The reason I'm making such a big deal out of this is that I figured I'd create an invariant to check that those fields remain unchanged. Maybe it would be worth adding another column "Read-only", for such a purpose.

Reply:
Love it!

Reply:
Done!